governance/program/src/processor/process_set_vote_authority.rs

```diff
@@ -0,0 +1,46 @@
+//! Program state processor
+
+use borsh::BorshSerialize;
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+};
+
+use crate::{
+    error::GovernanceError,
+    state::voter_record::{deserialize_voter_record, get_voter_record_address_seeds},
+};
+
+/// Processes SetVoteAuthority instruction
+pub fn process_set_vote_authority(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    realm: &Pubkey,
+    governing_token_mint: &Pubkey,
+    vote_authority: &Pubkey,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let governing_token_owner_info = next_account_info(account_info_iter)?; // 0
+    let voter_record_info = next_account_info(account_info_iter)?; // 1
+
+    if !governing_token_owner_info.is_signer {
```

Comment:
If someone is the voting authority, only the owner can ever reassign it at the moment. Does it make sense to allow the voting authority to also reassign their privileges?  That way the owner doesn't need to do anything if some management changes

Reply:
Agree, if we trust 3rd party to vote for us then it should also be ok to trust them to change `vote_authority`.
I've made the change and also set `vote_authority` to be None by default.